diesel_cli/src/main.rs

```diff
@@ -186,6 +188,59 @@ fn convert_absolute_path_to_relative(target_path: &Path, mut current_path: &Path
     result.join(target_path.strip_prefix(current_path).unwrap())
 }
 
+fn run_infer_schema(matches: &ArgMatches) {
+    let database_url = database::database_url(matches);
+    let schema_name = None;
+    let schema = diesel_infer_schema::infer_schema_for_schema_name(&database_url, schema_name);
+    let mut out = String::new();
+    let mut indent = String::new();
+    let mut skip_space = false;
+    let mut last_char = ' ';
+    
+
+    for c in schema.as_str().chars() {
+        if c == '!' || c == '(' || c == ';' || c == ','  || c == '<'
+            || (c == ':' && last_char != ':') ||
+            (c == '>' && last_char != '-') {
+            out.pop();
+        }
```

Comment:
Whew, this looks complicated. Can add a comment describing what is happening here? Is it possible to rewrite this whole block as an iterator chain that might be easier to read?